Add FSI section to Input File docs. (#1220)
Exawind/nalu-wind: docs/source/user/nalu_run/nalu_inp.rst

   Real vector indicating the location of the base of the tower.
   This should match the variable with the same name in the OpenFAST
   driver input file (with extension .yaml).

.. inpfile:: turbine_name.turbine_hub_pos

   Real vector indicating the location of the hub.
   This should match the variable with the same name in the OpenFAST
   driver input file (with extension .yaml).  Note that any height
   change due to rotor tilt should be included in this variable, but
   the effect of yaw should not be included.

.. inpfile:: turbine_name.restart_filename

   String (in double quotes) indicating the OpenFAST checkpoint file
   that the FSI simulation will start from.  Do not include the
   extension (.chkp).  This variable will have the format
   ``"name.integer"``, where the ``integer`` is the number of steps taken
   in the OpenFAST standalone simulation.  This should be equal to
   ``t_end`` divided by ``dt_FAST`` from the OpenFAST standalone run.

.. inpfile:: turbine_name.sim_type

   String (in double quotes) indicating the type of OpenFAST simulation.
   For FSI simulations, this should always be set to ``"ext-loads"``.

.. inpfile:: turbine_name.blade_parts

>    List of strings (in double quotes) indicating the mesh element blocks

Reviewer: Double quotes are not required for yaml to recognize strings.  Raw characters are fine.

   corresponding to each of the blades.

.. inpfile:: turbine_name.tower_parts

   List of strings (in double quotes) indicating the mesh element blocks
   corresponding to the tower.

.. inpfile:: turbine_name.blade_boundary_parts

   List of strings (in double quotes) indicating the mesh sidesets
   corresponding to the wall boundaries of each of the blades.

.. inpfile:: turbine_name.tower_boundary_parts

   List of strings (in double quotes) indicating the mesh sidesets
   corresponding to the wall boundaries of the tower.

.. inpfile:: turbine_name.az_blend_mean

   Real variable (in radians) indicating the mean angular position for the load blending.
   The loads provided to BeamDyn are a weighted average of the loads

>    provided by AeroDyn and the true CFD loads provided by Nalu-Wind.

Reviewer: ```suggestion
   provided by AeroDyn and the true CFD loads provided by Nalu-Wind during startup to help with stability while the CFD flow field develops around the turbine structure.
```

   The weight of the Nalu-Wind contribution takes the form
   :math:`\frac{1}{2} \left[1+\text{tanh}\left(\left(\phi-\phi_\text{mean}\right)/\phi_\text{delta}\right)\right]`,

>    where :math:`\phi` is the angle that the turbine has rotated through,

Reviewer: ```suggestion
   where :math:`\phi` is the angle that the turbine has rotated through (including any initial OpenFAST runs),
```

   :math:`\phi_\text{mean}` is the value of ``az_blend_mean``, and
   :math:`\phi_\text{delta}` is the value of ``az_blend_delta``.

.. inpfile:: turbine_name.az_blend_delta

>    Real variable (in radians) indicating the width for load blending.
>    The loads provided to BeamDyn are a weighted average of the loads
>    provided by AeroDyn and the true CFD loads provided by Nalu-Wind.
>    The weight of the Nalu-Wind contribution takes the form
>    :math:`\frac{1}{2} \left[1+\text{tanh}\left(\left(\phi-\phi_\text{mean}\right)/\phi_\text{delta}\right)\right]`,
>    where :math:`\phi` is the angle that the turbine has rotated through,
>    :math:`\phi_\text{mean}` is the value of ``az_blend_mean``, and
>    :math:`\phi_\text{delta}` is the value of ``az_blend_delta``.

Reviewer: ```suggestion
Definition supplied in the `turbine_name.az_blend_mean` parameter's description.
```
No need to repeat ourselves here 😎

   the wind is heading is measured clockwise from South.  For example, wind
   heading South is 0 degrees, and wind heading East is 270 degrees.  By convention,
   the x-axis points East, and the y-axis points North.
   This should match the variable with the same name in the OpenFAST driver input
   file (with extension .yaml).  If the turbine is intended to be aligned with
   the flow, the ``NacYaw`` variable found in the OpenFAST Elastodyn file
   should be 270 degrees minus ``wind_dir``.  For example, if the wind is heading East,
   ``NacYaw`` will be 0 degrees.

.. inpfile:: turbine_name.z_ref

   Real variable indicating the reference height at which ``vel_mean`` and
   ``wind_dir`` apply.  Often corresponds to the hub height of the turbine.

.. inpfile:: turbine_name.shear_exp

   Real variable indicating the exponent used in a power-law approximation
   of the incoming ABL.  This variable should match the variable with the same
   name in the OpenFAST driver input file (with extension .yaml).

.. inpfile:: turbine_name.deflection_ramping

   The ``deflection_ramping`` sub-subsection controls the temporal and spatial

>    ramping of blade deflections applied in Nalu.  The temporal ramping enables a smooth

Reviewer: ```suggestion
   ramping of blade deflections applied in Nalu-Wind.  The temporal ramping enables a smooth
```

>    transition from the reference configuration to the full deflections, and

Reviewer: ```suggestion
   transition for the blades from a rigid body motion based on the hub motion to the full blade deflections, and
```

>    improves the stability of the simulation.  The span ramping enables the root section

Reviewer: ```suggestion
   improves the stability of the simulation during startup.  The span ramping enables the root section
```

   improves the quality of the elements near the mesh interface between blades.
   Note that the total deflection ramping factor will be the product of the temporal,
   theta, and span ramping factors.  This sub-subsection should be provided for each turbine.

.. inpfile:: deflection_ramping.enable_temporal_ramping

   Boolean variable indicating whether temporal deflection ramping should be used.

.. inpfile:: deflection_ramping.enable_theta_ramping

   Boolean variable indicating whether spatial deflection ramping with respect to
   the circumferential direction should be used.

.. inpfile:: deflection_ramping.enable_span_ramping

   Boolean variable indicating whether spatial deflection ramping with respect to
   the spanwise direction should be used.

.. inpfile:: deflection_ramping.span_ramp_distance

   Real variable indicating the distance from the root over which to ramp deflections
   if ``enable_span_ramping`` is set to true.  Distance is measured in the same units
   as the mesh (meters in most cases).

.. inpfile:: deflection_ramping.temporal_ramp_start

   Real variable indicating the time at which temporal ramping should begin.

>    Prior to this time, zero deflections will be provided.  Note that this

Reviewer: ```suggestion
   Prior to this time, zero blade deflections will be provided.  Note that this
```

   should correspond to time as recorded by Nalu-Wind and described in the
   ``Time_Integrators`` section, not as recorded by OpenFAST.  Hence, this will
   not correspond to ``t_start`` above.

.. inpfile:: deflection_ramping.temporal_ramp_end

   Real variable indicating the time at which temporal ramping should end.

>    After this time, full deflections will be provided.  Note that this

Reviewer: ```suggestion
   After this time, full blade deflections will be provided.  Note that this
```
